oneko: 1.2.sakura.5 -> 1.2.hanami.6, and change upstream

IreneKnapp opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
As of approximately a year ago, I am the upstream maintainer of Oneko. The previous version was released in 1999. Per the README, every version has been by a different maintainer. I did attempt to contact the previous one, but since there was no response, I think it's entirely in the spirit of the project to consider my version the latest. I like to think of it as similar to those piles of stones, where each hiker who passes adds one more.

###### Things done
Update Oneko to version 1.2.hanami.6. Add myself as a second nixpkgs maintainer for it.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

xaverdh commented:
So while I very much appreciate the work, the license part makes this pr somewhat questionable.
I think some people might not want unfree (even if it is redistributable) software on their systems, so I think that we will want to either
* keep both the original and the new one, so ppl can choose
* or you may want to change the license of the new version to something like [gpl,mit,bsd,..] (not sure what your stance on this is)

IreneKnapp commented:
Yeah, so I added that not because I've changed anything but because I think it was incorrectly classified to begin with. It's one of those strange copyright scenarios that were common in the 90s, before everyone got used to digital rights stuff. My code additions are released into the public domain, but unfortunately the software has always contained various trademarked and copyrighted characters. The README documents that the owners of the art have given permission to use it, but I don't think it's properly described as free.

If I'm being too nitpicky, I'm happy to drop the unfreeRedistributable and let things to back to the status quo of not being examined too closely... I think this program is culturally significant and I don't want to cause trouble for it by asking inconvenient questions.

xaverdh commented:
Ah, thanks for explaining the reasoning!
debian classifies it as public domain (https://metadata.ftp-master.debian.org/changelogs//main/o/oneko/oneko_1.2.sakura.6-14_copyright) and they are usually *very* careful with these kind of things, so it should be fine I think.

IreneKnapp commented:
I appreciate your diligence. I put this together a while ago and had forgotten I added that part. Sorry about the confusion. Anyway, I set it back to just publicDomain; please take a look. :)

xaverdh reviewed (approved):
looks good!
You probably want to squash some of the commits before this will be merged.

SuperSandro2000 commented on pkgs/applications/misc/oneko/default.nix:
```diff
@@ -1,12 +1,14 @@
+  src = fetchFromGitHub {
+    owner = "IreneKnapp";
+    repo = "oneko";
+    rev = "${version_name}";
```
```suggestion
    rev = version_name;
```

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -4323,6 +4323,16 @@
     githubId = 54999;
     name = "Ariel Nunez";
   };
+  irenes = {
```
Please put this into a separate commit with the message ``maintainers: add irenes`` and sort it alphabetically.

IreneKnapp replied:
done; please take a look.

IreneKnapp committed:
maintainers: add irenes

IreneKnapp committed:
oneko: 1.2.sakura.5 -> 1.2.hanami.6; new upstream; add myself as maintainer

SuperSandro2000 commented:
@ofborg eval
